src/field/arena.ts

```diff
@@ -292,123 +175,32 @@ export class Arena {
-      this.trainerPool[BiomePoolTier.BOSS].length > 0
-      && (globalScene.gameMode.isTrainerBoss(waveIndex, this.biomeId) || isBoss);
-
-    // @todo Right now there are no super/ultra or rare boss trainers
-    const tierValue = randSeedInt(isTrainerBoss ? 64 : 512);
-    let tier = isTrainerBoss ? this.generateBossBiomeTier(tierValue) : this.generateNonBossBiomeTier(tierValue);
-
-    while (tier > BiomePoolTier.COMMON && !this.trainerPool[tier].length) {
-      console.log(
-        `Downgraded trainer rarity tier from ${enumValueToKey(BiomePoolTier, tier)} to ${enumValueToKey(BiomePoolTier, (tier - 1) as BiomePoolTier)}`,
-      );
-      tier--;
-    }
-    const tierPool = this.trainerPool[tier] || [];
-    return tierPool.length ? tierPool[randSeedInt(tierPool.length)] : TrainerType.BREEDER;
-  }
+  // #endregion
+  // #region Weather
 
   /**
-   * Generates a `formIndex` for a given species based on the biome and time of day.
-   * Used for Burmy/Wormadam, Rotom, and Lycanroc
-   * @param species - The {@linkcode PokemonSpecies} being checked
-   * @returns the appropriate formIndex
-   */
-  getSpeciesFormIndex(species: PokemonSpecies): number {
-    switch (species.speciesId) {
-      case SpeciesId.BURMY:
-      case SpeciesId.WORMADAM:
-        switch (this.biomeId) {
-          case BiomeId.BEACH:
-            return 1;
-          case BiomeId.SLUM:
-            return 2;
-        }
-        break;
-      case SpeciesId.ROTOM:
-        switch (this.biomeId) {
-          case BiomeId.VOLCANO:
-            return 1;
-          case BiomeId.SEA:
-            return 2;
-          case BiomeId.ICE_CAVE:
-            return 3;
-          case BiomeId.MOUNTAIN:
-            return 4;
-          case BiomeId.TALL_GRASS:
-            return 5;
-        }
-        break;
-      case SpeciesId.LYCANROC: {
-        const timeOfDay = this.getTimeOfDay();
-        switch (timeOfDay) {
-          case TimeOfDay.DAY:
-          case TimeOfDay.DAWN:
-            return 0;
-          case TimeOfDay.DUSK:
-            return 2;
-          case TimeOfDay.NIGHT:
-            return 1;
-        }
-        break;
-      }
-    }
-
-    return 0;
-  }
-
-  /**
-   * Returns a background terrain color ratio for a given biome
-   * @returns 0, 131/180 or 1
+   * Determines if one the specified weather effects is set in the arena.
+   * Does **not** take into account weather suppression effects.
+   * @see {@linkcode Weather.isEffectSuppressed} to check if the weather effect is suppressed.
+   * @param weather - {@linkcode WeatherType} or array of {@linkcode WeatherType} to check against
+   * @returns `true` if the arena is of the specified weather, `false` otherwise
    */
-  getBgTerrainColorRatioForBiome(): number {
-    switch (this.biomeId) {
-      case BiomeId.SPACE:
-        return 1;
-      case BiomeId.END:
-        return 0;
-    }
-
-    return 131 / 180;
+  public hasWeather(weather: WeatherType | WeatherType[]): boolean {
```

Comment:
Consider making this a spread argument:

```suggestion
  public hasWeather(...weatherTypes: WeatherType[]): boolean {
```

Reply:
Yeah, I've got another PR lined up after this one to do some refactoring; I wanted to keep the non-reorganization changes to a minimum in this PR since it's difficult to review with all the code being moved around.

That's planned, and I already took care of the `TODO`s you mentioned as well.